OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -46,6 +48,10 @@ protected override void OnInside(Actor self)
 			foreach (var t in target.TraitsImplementing<INotifyInfiltrated>())
 				t.Infiltrated(target, self);
 
+			var exp = self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>();
+			if (exp != null && target.Owner != self.Owner)
```

Comment:
IMO this should either be unconditional (i.e. based on the valid stances for infiltration), or have its own set of stances.